pyro/mesh/patch.py

```diff
@@ -885,5 +885,174 @@ def do_demo():
     mydata.pretty_print("a")
 
 
-if __name__ == "__main__":
-    do_demo()
+# **c checking
+class PolarGrid(Grid2d):
+    """
+    the 2-d grid class.  The grid object will contain the coordinate
+    information (at various centerings).
+
+    A basic (1-d) representation of the layout is::
+
+       |     |      |     X     |     |      |     |     X     |      |     |
+       +--*--+- // -+--*--X--*--+--*--+- // -+--*--+--*--X--*--+- // -+--*--+
+          0          ng-1    ng   ng+1         ... ng+nx-1 ng+nx      2ng+nx-1
+
+                            ilo                      ihi
+
+       |<- ng guardcells->|<---- nx interior zones ----->|<- ng guardcells->|
+
+    The '*' marks the data locations.
+    """
+
+    # pylint: disable=too-many-instance-attributes
+
+    def __init__(self, nx, ny, ng=1,
+                 xmin=0.0, xmax=1.0, ymin=0.0, ymax=1.0):
```

Comment:
I don't think you need __init__, since it looks the same as the base class

Reply:
:)

Reply:
I left if for later if it will be needed to be changed.

Reply:
It might be better to call `super().__init__(nx, ny, ...)` in that case